ADD l10n_es_aeat_mod111

JuanjoA opened the pull request:
Se añade l10n_es_aeat_mod111 para continuar desarrollo

Commit:
ADD l10n_es_aeat_mod111

Se añade l10n_es_aeat_mod111 para continuar desarrollo

pedrobaeza commented:
Perfecto. Etiqueto el PR como WIP (work in progress). Cualquier progreso, poned un comentario aquí.

Commit:
Editada vista para ajustes en campos

Añadido domain multicompañía para selección de fiscalyear. Añadido
campo period_ids con dominio para poder seleccionar el periodo adecuado
al año fiscal seleccionado (y por lo tanto a la compañía correcta)

Commit:
Varios cambios, se obtienen 07,08,09

El módulo instala ok, se obtienen los valores para las casillas 07, 08
y 09 (rendimientos actividades económicas)
Parece que las casillas 07 y 09 se salen ok, el valor de la 08 no lo
tengo tan claro.
Toda la lógica se apoya en apuntes contables

pedrobaeza commented on .gitignore:
```diff
@@ -54,3 +54,6 @@ docs/_build/
 *.swp
+
+TODO.txt
+.DS_Store
```
Y estos cambios en el .gitignore?

pedrobaeza commented on l10n_es_aeat_mod111/models/mod111.py:
```diff
@@ -0,0 +1,220 @@
+            'Apellidos o razón social', size=30, readonly=True,
+            states={'draft': [('readonly', False)]}, required=True),
+        'calculation_date': fields.datetime('Fecha de cálculo'),
+        'casilla_01': fields.integer('Casilla [01]', readonly=True),
```
En el help de cada campo se podría explicar con copia/pega las instrucciones del modelo para esa casilla.

pedrobaeza commented on l10n_es_aeat_mod111/models/mod111.py:
```diff
@@ -0,0 +1,220 @@
+            help='Código electrónico de la declaración anterior'
+            ' (si se presentó telemáticamente).'
+            ' A cumplimentar sólo en el caso de declaración complementaria.'),
+        'company_id': fields.many2one(
```
Creo que este campo, el company_vat, y algún otro no hace falta al estar ya en el modelo abstracto del módulo l10n_es_aeat

pedrobaeza commented on l10n_es_aeat_mod111/models/mod111.py:
```diff
@@ -0,0 +1,220 @@
+            help='Número de justificante de la declaración anterior'
+            ' (si se presentó en papel).'
+            ' A cumplimentar sólo en el caso de declaración complementaria.'),
+        'period_id': fields.many2one(
```
Se podría poner periodo inicio y fin como con otros modelos.

pedrobaeza commented on l10n_es_aeat_mod111/models/mod111.py:
```diff
@@ -0,0 +1,220 @@
+        'period_id': fields.many2one(
+            'account.period', 'Periodo', readonly=True,
+             states={'draft': [('readonly', False)]}, required=True),
+        'state': fields.selection(
```
Esta campo tampoco hace falta.

pedrobaeza commented on l10n_es_aeat_mod111/models/mod111.py:
```diff
@@ -0,0 +1,220 @@
+            states={'draft': [('readonly', False)]}, required=True),
+    }
+
+    def _check_complementary(self, cr, uid, ids, context=None):
```
Esto diría que tampoco hace falta

pedrobaeza commented on l10n_es_aeat_mod111/models/mod111.py:
```diff
@@ -0,0 +1,220 @@
+    ]
+
+    _defaults = {
+        'company_id': lambda self, cr, uid, context=None:
```
No hace falta

pedrobaeza commented on l10n_es_aeat_mod111/models/mod111.py:
```diff
@@ -0,0 +1,220 @@
+        'company_id': lambda self, cr, uid, context=None:
+            (self.pool['res.company']._company_default_get(
+                cr, uid, 'l10n.es.aeat.mod111.report', context=context)),
+        'complementaria': False,
```
No hace falta

pedrobaeza commented on l10n_es_aeat_mod111/models/mod111.py:
```diff
@@ -0,0 +1,220 @@
+                cr, uid, 'l10n.es.aeat.mod111.report', context=context)),
+        'complementaria': False,
+        'number': '111',
+        'state': 'draft',
```
No hace falta

pedrobaeza commented on l10n_es_aeat_mod111/models/mod111.py:
```diff
@@ -0,0 +1,220 @@
+
+
+
+    def get_account_child_ids(self, cr, uid, cuenta, company_id):
```
Existe ya un método en el estándar en account.account que hace esto. Creo que se llamaba _get_children_and_consol

pedrobaeza commented on l10n_es_aeat_mod111/models/mod111.py:
```diff
@@ -0,0 +1,220 @@
+        move_line_model = self.pool.get('account.move.line')
+        journal_model = self.pool.get('account.journal')
+
+        cuentas_hijas_4751 = self.get_account_child_ids(cr, uid, '4751', company_id.id)
```
No se debería harcodear la 4751, si no escoger los movimientos en base a ciertos códigos contables.

pedrobaeza commented:
Cierro este PR puesto que se ha hecho para la v8
